### PR TITLE
Track design fixes

### PR DESCRIPTION
**Pull request by janisozaur**
Track design from #5931 now costs 7 274 money to build, instead of -214 740.60.

I'm not sure if this is realistic in-game price for such a coaster.

### CODE REVIEW

**Comment by janisozaur** [comment]
dismaland 2017-08-01 07-37-57

File: src/openrct2/object.h [hunk 1]
```diff
@@ -65,7 +65,10 @@ typedef enum
  * size: 0x10
  */
 typedef struct rct_object_entry {
-    uint32 flags;
+    union {
+        uint8 end_flag; // needed not to read past allocated buffer.
```
Comment (Gymnasiast) [inline src/openrct2/object.h]: Why did you introduce this? Below you're still &-ing with 0xFF, so it doesn't seem to change anything.

Comment (janisozaur) [inline src/openrct2/object.h]: But it does. the `flags` field is 4 bytes wide and generates an appropriate read/write, but you only allocate space for one more byte when saving or loading, so you end up walking past available memory. With this code you only read single byte, which stays within bounds. Indeed, `& 0xFF` is now superfluous, so I will remove it.

**Comment by jensj12** [comment]
With scenery, the coaster should cost 7652, 7274 is the correct price without scenery. Below a screenshot of the same coaster, but saved and placed again.
happy park 2017-08-01 11-02-43

**Comment by janisozaur** [comment]
@jensj12 oh that's nice. How do you know?

**Comment by jensj12** [comment]
I placed the bugged coaster and saved it again. Then everything works as expected.

File: src/openrct2/ride/track_design.c [hunk 2]
```diff
@@ -895,7 +895,7 @@ static sint32 track_design_place_scenery(rct_td6_scenery_element *scenery_start,
                     continue;
                     break;
                 }
-                _trackDesignPlaceCost += cost;
+                add_clamp_money32(_trackDesignPlaceCost, cost);
```
Comment (IntelOrca) [inline src/openrct2/ride/track_design.c]: Is this line the actual fix for the issue? Everything else seems unrelated.

Comment (janisozaur) [inline src/openrct2/ride/track_design.c]: Not exactly unrelated, but yes, that's the crux of the matter
